hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java

```diff
@@ -110,7 +109,7 @@ public Collection<OzoneFileStatus> listStatusFSO(OmKeyArgs args,
     // if the keyName is null
     if (StringUtils.isNotBlank(startKey)) {
       if (StringUtils.isNotBlank(keyName)) {
-        if (!listKeysMode &&
+        if (!allowPartialPrefixes &&
             !OzoneFSUtils.isImmediateChild(keyName, startKey)) {
```

Comment:
Clarification Point: Here its validating `isImmediateChild(keyName, startKey) `only if the allowPartialPrefixes=false. What if the partial startKey prefix is not an immediate child ?